metricbeat/helper/sql/sql.go

```diff
@@ -44,22 +44,25 @@ type sqlRow interface {
 
 // NewDBClient gets a client ready to query the database
 func NewDBClient(driver, uri string, l *logp.Logger) (*DbClient, error) {
-	dbx, err := sql.Open(switchDriverName(driver), uri)
+	dbx, err := sql.Open(SwitchDriverName(driver), uri)
 	if err != nil {
 		return nil, fmt.Errorf("opening connection: %w", err)
 	}
 	err = dbx.Ping()
 	if err != nil {
 		if closeErr := dbx.Close(); closeErr != nil {
-			return nil, fmt.Errorf("failed to close with %s, after connection test failed: %w", closeErr, err)
+			// NOTE(SS): Support for wrapping multiple errors is there in Go 1.20+.
+			// TODO(SS): When beats module starts using Go 1.20+, use: https://pkg.go.dev/errors#Join
+			// and until then, let's use the following workaround.
+			return nil, fmt.Errorf(fmt.Sprintf("failed to close with: %s", closeErr.Error())+" after connection test failed: %w", err)
```

Comment:
Is `fmt.Sprintf` necessary here?

Reply:
Had to fix it because it was pointed out by the linter.

And yes, it is necessary. As you can see from the comments, multiple errors cannot be wrapped. That feature is there in Go 1.20+ i.e., very soon we can replace this but not as of now. So, did a workaround be making one error a string and wrapped the other one.

Reply:
https://pkg.go.dev/errors#Join

Reply:
`return nil, fmt.Errorf("failed to close with: %s after connection test failed: %w", closeErr.Error(), err)` this should be the same thing without `fmt.Sprintf`.

Reply:
Yes. You are right. But linter that's raising the issue is dumb. So, did the same thing in a different way to confuse the linter :sweat_smile:

Reply:
closeError -> closeError.Error() can also fix this as the latter is not an error value but a string; hence, errorlint doesn't think it should be wrapped using %w.